index.js

```diff
@@ -212,24 +241,58 @@ NYC.prototype.report = function (cb, _collector, _reporter) {
   reporter.write(collector, true, cb)
 }
 
+var loadedMaps = {}
+
 NYC.prototype._loadReports = function () {
   var _this = this
   var files = fs.readdirSync(this.tmpDirectory())
 
-  return _.map(files, function (f) {
+  var SourceMapCache = require('./lib/source-map-cache')
+  var sourceMapCache = new SourceMapCache()
+
+  var cacheDir = _this.cacheDirectory()
+
+  return files.map(function (f) {
+    var report
     try {
-      return JSON.parse(fs.readFileSync(
+      report = JSON.parse(fs.readFileSync(
         path.resolve(_this.tmpDirectory(), './', f),
         'utf-8'
       ))
     } catch (e) { // handle corrupt JSON output.
       return {}
     }
+
+    if (report) {
```

Comment:
This condition seems unnecessary. If the file no longer exists, or did not contain valid JSON, we'll already have returned. The other option is that it contains valid JSON but it's not an object. There's plenty other ways the report could be invalid for the code that follows.